### PR TITLE
OSDOCS-4447: Add CNI doc

### PR DESCRIPTION
**Pull request by zshi-redhat**

### CODE REVIEW

**Review by mangelajo — CHANGES_REQUESTED** [review]
Some comments, nice work.

File: docs/default_cni_plugin.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+## Configurations
+
+### Configuring ovn-kubernetes
+
+The user provided ovn-kubernetes config should be written to the same directory as the MicroShift config.
+MicroShift will assume default ovn-kubernetes config values if ovn-kubernetes config file is not provided.
+These paths will be checked in order for the ovn-kubernetes config.
+
+1. User config dir: `~/.microshift/ovn.yaml`
+2. Global config dir: `/etc/microshift/ovn.yaml`
```
[the remark this answers is not in the document]
Comment (zshi-redhat) [reply docs/default_cni_plugin.md]: Fixed.

File: docs/default_cni_plugin.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+When `disableOVSInit` is set to true in ovn-kubernetes CNI config file, OVS bridge "br-ex" needs to be manually configured:
+
+```bash
+sudo systemctl enable openvswitch --now
+sudo ovs-vsctl add-br br-ex
+sudo ovs-vsctl add-port br-ex <physical-interface-name>
+sudo ip link set br-ex up
+```
+
```
[the remark this answers is not in the document]
Comment (zshi-redhat) [reply docs/default_cni_plugin.md:94]: Added TODO comment to replace ovs cmds with nmcli (comment is only visible in text).

File: docs/default_cni_plugin.md [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+A wide range of networking features are available with ovn-kubernetes, including but not limited to:
+
+* Kubernetes network policy
+* Dynamic node IP
+* Cluster network on specified host interface
+* Egress via additional host interface
+* Dual stack
+* Service idling (disabled)
+* Egress firewall/QOS/IP (disabled)
+* East-west multicast
```
[the remark this answers is not in the document]
Comment (zshi-redhat) [reply docs/default_cni_plugin.md]: Removed multicast.

File: docs/default_cni_plugin.md [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+### Cluster network on specified host interface
+
+microshift-ovs-init.service is able to use user specified host interface for cluster network.
+This is done by specifying the desired host interface name in a hint file `/var/lib/ovnk/iface_default_hint`.
```
[the remark this answers is not in the document]
Comment (zshi-redhat) [reply docs/default_cni_plugin.md:204]: I will update this once we add support in configure-ovs.sh script.

**Review by mangelajo — APPROVED** [review]
/lgtm

**Pull request merged**